pagerduty/extension.go

```diff
@@ -0,0 +1,95 @@
+package pagerduty
+
+import "fmt"
+
+// ExtensionService handles the communication with extension related methods
+// of the PagerDuty API.
+type ExtensionService service
+
+// Extension represents an extension.
+type Extension struct {
+	Extension        *Extension                `json:"extension,omitempty"`
```

Comment:
Why is there a pointer to the struct being defined? Can an extension link to another extension?

I've noticed this was done on most struct definitions in `go-pagerduty` but I don't see how this matches with the API.

Reply:
Hi @pdecat, this is merely a work-around used internally within the client to marshal/unmarshal requests/responses, which I'm planning to replace with proper request/response structs in the near future.